web/app/view/CustomNumberField.js

```diff
@@ -20,30 +20,79 @@ Ext.define('Traccar.view.CustomNumberField', {
     xtype: 'customNumberField',
 
     beforeEl: '<div style="width:100%;display:inline-table;">',
-    unitEl: '<div style="display:table-cell;padding-left:10px;vertical-align:middle;">',
+    unitEl: '<div id="numberUnitEl" style="display:table-cell;padding-left:10px;vertical-align:middle;width:1px;white-space:nowrap;">',
 
     constructor: function (config) {
-        var unit;
-        if (config.dataType === 'speed') {
-            unit = Traccar.app.getPreference('speedUnit', 'kn');
-            config.beforeSubTpl = this.beforeEl;
-            config.afterSubTpl = this.unitEl + Ext.getStore('SpeedUnits').findRecord('key', unit).get('name') + '</div></div>';
+        var unitName = '';
+        if (config.dataType) {
+            config.beforeBodyEl = this.beforeEl;
```

Comment:
Make wrapping `div` around only body, not whole component
![image](https://user-images.githubusercontent.com/5688080/30046844-0d87588e-9227-11e7-9291-20e94976babc.png)